### DISCUSSION TITLE
how pnpm treat ^ in peerDependencies?

### DISCUSSION THREAD

**Discussion by ivy1201** [pnpm/pnpm#4090] [Q&A]
suppose app rely's on package A, and B is defined in A's peerDependencies like:
peerDependencies: {
     B: "^1.0.1"
}
and B's latest version is 1.1.2
but B's actual version installed in the app  is 1.0.1, seems ^ is not workable

**Comment by zkochan** [comment]
pnpm will install B@1.0.1 and print a warning

**Reply by ivy1201** [reply]
but i want to install 1.1.2 actually in my case, is there any way to achieve this effect?

**Reply by zkochan** [reply]
you may use an override to force version 1.1.2 in B

https://pnpm.io/package_json#pnpmoverrides

**Reply by ivy1201** [reply]
seems not work in rush project with pnpm as the package manager?

**Reply by ivy1201** [reply]
i am confused why ^ in peerDependencies  do not work as in dependencies to match the major version, it is by design?

**Reply by zkochan** [reply]
oh, the peer is `^1.0.1`, then it should work as you expect.
